data-prepper-plugins/saas-source-plugins/source-crawler/src/main/java/org/opensearch/dataprepper/plugins/source/source_crawler/auth/AuthenticationInterface.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.dataprepper.plugins.source.source_crawler.auth;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+
+/**
+ * Generic authentication interface for SaaS source plugins.
+ * Provides common authentication functionality including credential initialization and renewal.
+ */
+public interface AuthenticationInterface {
+    Logger log = LoggerFactory.getLogger(AuthenticationInterface.class);
+
+    /**
+     * Gets the retry interval in minutes for credential initialization.
+     * This method can be overridden by implementing classes to customize the retry interval.
+     * @return the retry interval in minutes (default is 10 minutes)
+     */
+    default long getRetryIntervalMinutes() {
+        return 10;
+    }
+
+    /**
+     * Initializes credentials with retry logic.
+     * This method will attempt to renew credentials and retry on failure.
+     */
+    default void initCredentials() {
+        log.info("Initializing credentials.");
+        while (!isCredentialsInitialized()) {
+            try {
+                renewCredentials();
+                setCredentialsInitialized(true);
+                log.info("Credentials initialized successfully");
+                break;
+            } catch (HttpClientErrorException | HttpServerErrorException | SecurityException ex) {
```

Comment:
You should also have a catch-all or you may leave your loop. That catch all would be catching unexpected exceptions and could include the stack tracke.